src/bgen/TypeManager.cs

```diff
@@ -1,11 +1,15 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
 using System.Reflection;
 
 #nullable enable
 
 public class TypeManager {
+	public BindingTouch BindingTouch;
+	readonly HashSet<string> typesThatMustAlwaysBeGloballyNamed = new ();
```

Comment:
Nothing seems to ever add to this hash set?

Reply:
Ah, I forgot to bring over that logic from the Generator. Added in new commit.